---
title: "Repeat elements"
output:
  html_document:
    toc: true
    toc_float: true
    number_sections: true
---

# Setup

Working directory on EBI cluster: `/hps/research1/birney/users/ian/mikk_paper`

## Create `conda` environments

### For Linux

```{bash, eval = F}
# Create and activate environment 
conda create -n repeat_env
conda activate repeat_env
# Add packages
conda install -c bioconda repeatmasker 
conda install -c bioconda bcftools 
# Export to repo
conda env export > mikk_genome/code/config/repeat_env.yml
```

### For R

```{bash, eval = F}
conda create -n r
conda activate r
conda install -c conda-forge r-base
conda install -c conda-forge r-here 
conda install -c r r-tidyverse
conda install -c bioconda bioconductor-genomicranges
conda install -c conda-forge r-scales
# Export to repo
conda env export > mikk_genome/code/config/r.yml
```

## Get Jack's data

Jack's directory here: `/nfs/leia/research/enright/jack/10_medaka_fish/02_genome_repeats/repeatmasker_filtered/medaka_hdrr`

```{bash, eval = F}
# Create directory for repeats work
mkdir repeats
# Pull data over from Jack's repo via local
scp brettell@yoda:/nfs/leia/research/enright/jack/10_medaka_fish/02_genome_repeats/repeatmasker_filtered/medaka_hdrr/processed/medaka_hdrr_repeats.fixed.gff ~/Documents/Repositories/mikk_genome/data

scp ~/Desktop/medaka_hdrr_repeats.fixed.gff brettell@ebi:/hps/research1/birney/users/ian/mikk_paper/repeats
```

# Create consensus sequence FASTA for each sample

```{bash, eval = F}
# NOTE: this is for when we want to analyse the repeats in each of the MIKK panel lines.
mkdir fastas_for_reps

full_panel_list=mikk_genome/data/20200210_panel_lines_full.txt 
ref=refs/Oryzias_latipes.ASM223467v1.dna.toplevel.fa
out_dir=fastas_for_reps
in_file=vcfs/full-run_line-ids.vcf.gz

for i in $( cat $full_panel_list ) ; do
  sample=$( echo $i | sed 's/-/_/g' ) ;
  bsub \
    -M 10000 \
    -o log/20201126_conc_$sample.out \
    -e log/20201126_conc_$sample.err \
    "bcftools consensus \
      --fasta-ref $ref \
      --sample $sample \
      --output $out_dir/$sample.fa \
      $in_file" ;
done      
```

# Process data

## Activate `r` environment

```{bash, eval = F}
conda activate r
R
```


## Load packages

```{r, message = F, warning = F}
require(here)
require(tidyverse)
require(GenomicRanges)
library(scales)
require(karyoploteR)
```


## Read in data and clean

```{r}
# Read in data
hdrr_reps = read.table(here("data", "20201126_medaka_hdrr_repeats.fixed.gff"),
                       header = F, sep = "\t", skip = 3, comment.char = "", quote = "", as.is = T) %>% 
  # Remove empty V8 column
  dplyr::select(-V8) %>% 
  # Get class of repeat from third column
  dplyr::mutate(class = stringr::str_split(V3, pattern = "#", simplify = T)[, 1]) %>% 
  # Rename columns
  dplyr::rename(chr = V1, tool = V2, class_full = V3, start = V4, end = V5, percent = V6, strand = V7, info = V9)

# Show head of table
knitr::kable(head(hdrr_reps, 10), format = "markdown")

# Find types of class other than "(GATCCA)n" types
class_types = unique(hdrr_reps$class[grep(")n", hdrr_reps$class, invert = T)])
class_types

# How many in the blank class?
length(which(hdrr_reps$class == ""))

# Recode class 
hdrr_reps = hdrr_reps %>% 
  # NA for blanks
  dplyr::mutate(class = dplyr::na_if(class, "")) %>% 
  # "misc" for others in "(GATCCA)n" type classes
  dplyr::mutate(class = dplyr::if_else(!class %in% class_types, "misc", class))

# Show head of new table
knitr::kable(head(hdrr_reps, 10), format = "markdown")

# Split into list based on class
rep_list = split(hdrr_reps, f = hdrr_reps$class)
```

## Convert into GenomicRanges

```{r}
rep_list_ranges = lapply(rep_list, function(x){
  ranges = GenomicRanges::makeGRangesFromDataFrame(x,
                                                   ignore.strand = T,
                                                   seqnames.field = "chr",
                                                   start.field = "start",
                                                   end.field = "end")
  return(ranges)
})
```

# Prepare data for karyoplot

## Get HdrR reference genome for karyoplot scaffold

```{r}
# Get chromosome lengths
med_chr_lens = read.table(here("data", "Oryzias_latipes.ASM223467v1.dna.toplevel.fa_chr_counts.txt"),
                          col.names = c("chr", "end"))
# Add start
med_chr_lens$start = 1
# Reorder
med_chr_lens = med_chr_lens %>% 
  dplyr::select(chr, start, end)
# Create custom genome
med_genome = regioneR::toGRanges(med_chr_lens)
```

## Load exon ranges

```{r}
# Load exon density for app
ex_ranges = readRDS(here("data", "20201125_exon_ranges.rds"))
```

## Make palette 

```{r}
electro_angler = c("#7400b8","#6930c3","#5e60ce","#5390d9","#4ea8de","#48bfe3","#56cfe1","#64dfdf","#72efdd","#80ffdb")
new_pal = grDevices::colorRampPalette(electro_angler)
```

# Generate karyoplot

## Full genome

**NOTE**: The y-scale for each class of repeat element is different.

```{r, fig.height=26, fig.width=17}
# Create Karyoplot
kp = plotKaryotype(med_genome, chromosomes = NULL)
# Add base numbers 
karyoploteR::kpAddBaseNumbers(kp, tick.dist = 5000000, minor.tick.dist = 1000000, add.units = T, cex = 0.5)

# Add mirror exon density to ideogram
kpPlotDensity(kp, data=ex_ranges, col = "#f77cb5",
              data.panel = "ideogram",
              window.size = 25000,
              r0 = 0.5, r1 = 1)
kpPlotDensity(kp, data=ex_ranges, col = "#f77cb5",
              data.panel = "ideogram",
              window.size = 25000,
              r0 = 0.5, r1 = 0)

# Set target classes
target_classes = c("DNA", "LINE", "LTR", "SINE", "misc")

# Get colours
cols = new_pal(length(target_classes))

# Create function to add densities to plot
for (i in seq_len(length(target_classes))){
  # get target_class
  target_class = target_classes[i]
  # get autotrack to automatically adjust r0 and r1
  at = autotrack(i, length(target_classes), r0=0, r1=1)
  # add densities
  kp = kpPlotDensity(kp, data = rep_list_ranges[[target_class]],
                     col = cols[i],
                     r0=at$r0, r1=at$r1,
                     window.size = 25000)
  computed_ymax = ceiling(kp$latest.plot$computed.values$max.density)
  # add axis
  kpAxis(kp, ymin=0, ymax=computed_ymax, numticks = 2,
         labels = c(NA, computed_ymax),
         r0=at$r0, r1=at$r1,
         cex = 0.5)
  # add labels
  kpAddLabels(kp, labels = target_class,
              r0=at$r0, r1=at$r1,
              cex = 0.6, label.margin = 0.035)
}
```

## Per chromosome

**NOTE**: As the plot for every chromosome is generated separately, the y-scale for the same class of repeat element changes with each plot. The chromosomes now also look as if they're the same length. It is therefore preferable to scan for genome-wide anomolies using the full-genome plot above.

```{r, fig.height=4, fig.width=15}
for (chr in as.character(seq(1, 24))){
  # Create Karyoplot
  kp = plotKaryotype(med_genome, chromosomes = chr)
  # Add base numbers 
  karyoploteR::kpAddBaseNumbers(kp, tick.dist = 5000000, minor.tick.dist = 1000000, add.units = T, cex = 0.5)
  
  # Add mirror exon density to ideogram
  kpPlotDensity(kp, data=ex_ranges, col = "#f77cb5",
                data.panel = "ideogram",
                window.size = 25000,
                r0 = 0.5, r1 = 1)
  kpPlotDensity(kp, data=ex_ranges, col = "#f77cb5",
                data.panel = "ideogram",
                window.size = 25000,
                r0 = 0.5, r1 = 0)
  
  # Set target classes
  target_classes = c("DNA", "LINE", "LTR", "SINE", "misc")
  
  # Get colours
  cols = new_pal(length(target_classes))
  
  # Create function to add densities to plot
  for (i in seq_len(length(target_classes))){
    # get target_class
    target_class = target_classes[i]
    # get autotrack to automatically adjust r0 and r1
    at = autotrack(i, length(target_classes), r0=0, r1=1)
    # add densities
    kp = kpPlotDensity(kp, data = rep_list_ranges[[target_class]],
                       col = cols[i],
                       r0=at$r0, r1=at$r1,
                       window.size = 25000)
    computed_ymax = ceiling(kp$latest.plot$computed.values$max.density)
    # add axis
    kpAxis(kp, ymin=0, ymax=computed_ymax, numticks = 2,
           labels = c(NA, computed_ymax),
           r0=at$r0, r1=at$r1,
           cex = 0.5)
    # add labels
    kpAddLabels(kp, labels = target_class,
                r0=at$r0, r1=at$r1,
                cex = 0.6, label.margin = 0.035)
  }
}  
```

