<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Repeat elements</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MIKK panel genome</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="20201120_markdown_ld-decay.html">Linkage Disequilibrium</a>
</li>
<li>
  <a href="20201125_markdown_introgression.html">Introgression</a>
</li>
<li>
  <a href="20201126_repeats.html">Repeat Elements</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Repeat elements</h1>

</div>


<div id="setup" class="section level1">
<h1><span class="header-section-number">1</span> Setup</h1>
<p>Working directory on EBI cluster: <code>/hps/research1/birney/users/ian/mikk_paper</code></p>
<div id="create-conda-environments" class="section level2">
<h2><span class="header-section-number">1.1</span> Create <code>conda</code> environments</h2>
<div id="for-linux" class="section level3">
<h3><span class="header-section-number">1.1.1</span> For Linux</h3>
<pre class="bash"><code># Create and activate environment 
conda create -n repeat_env
conda activate repeat_env
# Add packages
conda install -c bioconda repeatmasker 
conda install -c bioconda bcftools 
# Export to repo
conda env export &gt; mikk_genome/code/config/repeat_env.yml</code></pre>
</div>
<div id="for-r" class="section level3">
<h3><span class="header-section-number">1.1.2</span> For R</h3>
<pre class="bash"><code>conda create -n r
conda activate r
conda install -c conda-forge r-base
conda install -c conda-forge r-here 
conda install -c r r-tidyverse
conda install -c bioconda bioconductor-genomicranges
conda install -c conda-forge r-scales
# Export to repo
conda env export &gt; mikk_genome/code/config/r.yml</code></pre>
</div>
</div>
<div id="get-jacks-data" class="section level2">
<h2><span class="header-section-number">1.2</span> Get Jack’s data</h2>
<p>Jack’s directory here: <code>/nfs/leia/research/enright/jack/10_medaka_fish/02_genome_repeats/repeatmasker_filtered/medaka_hdrr</code></p>
<pre class="bash"><code># Create directory for repeats work
mkdir repeats
# Pull data over from Jack&#39;s repo via local
scp brettell@yoda:/nfs/leia/research/enright/jack/10_medaka_fish/02_genome_repeats/repeatmasker_filtered/medaka_hdrr/processed/medaka_hdrr_repeats.fixed.gff ~/Documents/Repositories/mikk_genome/data

scp ~/Desktop/medaka_hdrr_repeats.fixed.gff brettell@ebi:/hps/research1/birney/users/ian/mikk_paper/repeats</code></pre>
</div>
</div>
<div id="create-consensus-sequence-fasta-for-each-sample" class="section level1">
<h1><span class="header-section-number">2</span> Create consensus sequence FASTA for each sample</h1>
<pre class="bash"><code># NOTE: this is for when we want to analyse the repeats in each of the MIKK panel lines.
mkdir fastas_for_reps

full_panel_list=mikk_genome/data/20200210_panel_lines_full.txt 
ref=refs/Oryzias_latipes.ASM223467v1.dna.toplevel.fa
out_dir=fastas_for_reps
in_file=vcfs/full-run_line-ids.vcf.gz

for i in $( cat $full_panel_list ) ; do
  sample=$( echo $i | sed &#39;s/-/_/g&#39; ) ;
  bsub \
    -M 10000 \
    -o log/20201126_conc_$sample.out \
    -e log/20201126_conc_$sample.err \
    &quot;bcftools consensus \
      --fasta-ref $ref \
      --sample $sample \
      --output $out_dir/$sample.fa \
      $in_file&quot; ;
done      </code></pre>
</div>
<div id="process-data" class="section level1">
<h1><span class="header-section-number">3</span> Process data</h1>
<div id="activate-r-environment" class="section level2">
<h2><span class="header-section-number">3.1</span> Activate <code>r</code> environment</h2>
<pre class="bash"><code>conda activate r
R</code></pre>
</div>
<div id="load-packages" class="section level2">
<h2><span class="header-section-number">3.2</span> Load packages</h2>
<pre class="r"><code>require(here)
require(tidyverse)
require(GenomicRanges)
library(scales)
require(karyoploteR)</code></pre>
</div>
<div id="read-in-data-and-clean" class="section level2">
<h2><span class="header-section-number">3.3</span> Read in data and clean</h2>
<pre class="r"><code># Read in data
hdrr_reps = read.table(here(&quot;data&quot;, &quot;20201126_medaka_hdrr_repeats.fixed.gff&quot;),
                       header = F, sep = &quot;\t&quot;, skip = 3, comment.char = &quot;&quot;, quote = &quot;&quot;, as.is = T) %&gt;% 
  # Remove empty V8 column
  dplyr::select(-V8) %&gt;% 
  # Get class of repeat from third column
  dplyr::mutate(class = stringr::str_split(V3, pattern = &quot;#&quot;, simplify = T)[, 1]) %&gt;% 
  # Rename columns
  dplyr::rename(chr = V1, tool = V2, class_full = V3, start = V4, end = V5, percent = V6, strand = V7, info = V9)

# Show head of table
knitr::kable(head(hdrr_reps, 10), format = &quot;markdown&quot;)</code></pre>
<table>
<colgroup>
<col width="3%" />
<col width="11%" />
<col width="16%" />
<col width="5%" />
<col width="4%" />
<col width="6%" />
<col width="6%" />
<col width="37%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">chr</th>
<th align="left">tool</th>
<th align="left">class_full</th>
<th align="right">start</th>
<th align="right">end</th>
<th align="right">percent</th>
<th align="left">strand</th>
<th align="left">info</th>
<th align="left">class</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">DNA#P</td>
<td align="right">54</td>
<td align="right">131</td>
<td align="right">14.5</td>
<td align="left">-</td>
<td align="left">Target “Motif:rnd-4_family-856”</td>
<td align="left">DNA</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">Unknown</td>
<td align="right">85</td>
<td align="right">144</td>
<td align="right">16.7</td>
<td align="left">+</td>
<td align="left">Target “Motif:rnd-6_family-210”</td>
<td align="left">Unknown</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">SINE#tRNA-Core-RTE</td>
<td align="right">789</td>
<td align="right">866</td>
<td align="right">19.5</td>
<td align="left">+</td>
<td align="left">Target “Motif:rnd-6_family-2543”</td>
<td align="left">SINE</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">(CA)n</td>
<td align="right">921</td>
<td align="right">947</td>
<td align="right">12.2</td>
<td align="left">+</td>
<td align="left">Target “Motif:(CA)n”</td>
<td align="left">(CA)n</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">LINE#L2</td>
<td align="right">4464</td>
<td align="right">4596</td>
<td align="right">26.9</td>
<td align="left">+</td>
<td align="left">Target “Motif:DF0004089_LINE/L2” 959 1101"</td>
<td align="left">LINE</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">LINE#L2</td>
<td align="right">4478</td>
<td align="right">4599</td>
<td align="right">28.7</td>
<td align="left">+</td>
<td align="left">Target “Motif:DF0003433_LINE/L2” 1011 1141"</td>
<td align="left">LINE</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">(GAG)n</td>
<td align="right">6066</td>
<td align="right">6147</td>
<td align="right">26.8</td>
<td align="left">+</td>
<td align="left">Target “Motif:(GAG)n”</td>
<td align="left">(GAG)n</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">(AAATAA)n</td>
<td align="right">8329</td>
<td align="right">8360</td>
<td align="right">10.1</td>
<td align="left">+</td>
<td align="left">Target “Motif:(AAATAA)n”</td>
<td align="left">(AAATAA)n</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">(T)n</td>
<td align="right">8363</td>
<td align="right">8397</td>
<td align="right">23.5</td>
<td align="left">+</td>
<td align="left">Target “Motif:(T)n”</td>
<td align="left">(T)n</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">(TTTA)n</td>
<td align="right">8363</td>
<td align="right">8396</td>
<td align="right">12.8</td>
<td align="left">+</td>
<td align="left">Target “Motif:(TTTA)n”</td>
<td align="left">(TTTA)n</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Find types of class other than &quot;(GATCCA)n&quot; types
class_types = unique(hdrr_reps$class[grep(&quot;)n&quot;, hdrr_reps$class, invert = T)])
class_types</code></pre>
<pre><code>##  [1] &quot;DNA&quot;           &quot;Unknown&quot;       &quot;SINE&quot;          &quot;LINE&quot;          &quot;LTR&quot;           &quot;Simple_repeat&quot; &quot;Satellite&quot;    
##  [8] &quot;RC&quot;            &quot;tRNA&quot;          &quot;rRNA&quot;          &quot;Retroposon&quot;    &quot;snRNA&quot;         &quot;&quot;              &quot;buffer&quot;       
## [15] &quot;ARTEFACT&quot;      &quot;scRNA&quot;</code></pre>
<pre class="r"><code># How many in the blank class?
length(which(hdrr_reps$class == &quot;&quot;))</code></pre>
<pre><code>## [1] 8</code></pre>
<pre class="r"><code># Recode class 
hdrr_reps = hdrr_reps %&gt;% 
  # NA for blanks
  dplyr::mutate(class = dplyr::na_if(class, &quot;&quot;)) %&gt;% 
  # &quot;misc&quot; for others in &quot;(GATCCA)n&quot; type classes
  dplyr::mutate(class = dplyr::if_else(!class %in% class_types, &quot;misc&quot;, class))

# Show head of new table
knitr::kable(head(hdrr_reps, 10), format = &quot;markdown&quot;)</code></pre>
<table style="width:100%;">
<colgroup>
<col width="3%" />
<col width="11%" />
<col width="16%" />
<col width="5%" />
<col width="4%" />
<col width="7%" />
<col width="6%" />
<col width="38%" />
<col width="7%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">chr</th>
<th align="left">tool</th>
<th align="left">class_full</th>
<th align="right">start</th>
<th align="right">end</th>
<th align="right">percent</th>
<th align="left">strand</th>
<th align="left">info</th>
<th align="left">class</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">DNA#P</td>
<td align="right">54</td>
<td align="right">131</td>
<td align="right">14.5</td>
<td align="left">-</td>
<td align="left">Target “Motif:rnd-4_family-856”</td>
<td align="left">DNA</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">Unknown</td>
<td align="right">85</td>
<td align="right">144</td>
<td align="right">16.7</td>
<td align="left">+</td>
<td align="left">Target “Motif:rnd-6_family-210”</td>
<td align="left">Unknown</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">SINE#tRNA-Core-RTE</td>
<td align="right">789</td>
<td align="right">866</td>
<td align="right">19.5</td>
<td align="left">+</td>
<td align="left">Target “Motif:rnd-6_family-2543”</td>
<td align="left">SINE</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">(CA)n</td>
<td align="right">921</td>
<td align="right">947</td>
<td align="right">12.2</td>
<td align="left">+</td>
<td align="left">Target “Motif:(CA)n”</td>
<td align="left">misc</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">LINE#L2</td>
<td align="right">4464</td>
<td align="right">4596</td>
<td align="right">26.9</td>
<td align="left">+</td>
<td align="left">Target “Motif:DF0004089_LINE/L2” 959 1101"</td>
<td align="left">LINE</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">LINE#L2</td>
<td align="right">4478</td>
<td align="right">4599</td>
<td align="right">28.7</td>
<td align="left">+</td>
<td align="left">Target “Motif:DF0003433_LINE/L2” 1011 1141"</td>
<td align="left">LINE</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">(GAG)n</td>
<td align="right">6066</td>
<td align="right">6147</td>
<td align="right">26.8</td>
<td align="left">+</td>
<td align="left">Target “Motif:(GAG)n”</td>
<td align="left">misc</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">(AAATAA)n</td>
<td align="right">8329</td>
<td align="right">8360</td>
<td align="right">10.1</td>
<td align="left">+</td>
<td align="left">Target “Motif:(AAATAA)n”</td>
<td align="left">misc</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">(T)n</td>
<td align="right">8363</td>
<td align="right">8397</td>
<td align="right">23.5</td>
<td align="left">+</td>
<td align="left">Target “Motif:(T)n”</td>
<td align="left">misc</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">RepeatMasker</td>
<td align="left">(TTTA)n</td>
<td align="right">8363</td>
<td align="right">8396</td>
<td align="right">12.8</td>
<td align="left">+</td>
<td align="left">Target “Motif:(TTTA)n”</td>
<td align="left">misc</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Split into list based on class
rep_list = split(hdrr_reps, f = hdrr_reps$class)</code></pre>
</div>
<div id="convert-into-genomicranges" class="section level2">
<h2><span class="header-section-number">3.4</span> Convert into GenomicRanges</h2>
<pre class="r"><code>rep_list_ranges = lapply(rep_list, function(x){
  ranges = GenomicRanges::makeGRangesFromDataFrame(x,
                                                   ignore.strand = T,
                                                   seqnames.field = &quot;chr&quot;,
                                                   start.field = &quot;start&quot;,
                                                   end.field = &quot;end&quot;)
  return(ranges)
})</code></pre>
</div>
</div>
<div id="prepare-data-for-karyoplot" class="section level1">
<h1><span class="header-section-number">4</span> Prepare data for karyoplot</h1>
<div id="get-hdrr-reference-genome-for-karyoplot-scaffold" class="section level2">
<h2><span class="header-section-number">4.1</span> Get HdrR reference genome for karyoplot scaffold</h2>
<pre class="r"><code># Get chromosome lengths
med_chr_lens = read.table(here(&quot;data&quot;, &quot;Oryzias_latipes.ASM223467v1.dna.toplevel.fa_chr_counts.txt&quot;),
                          col.names = c(&quot;chr&quot;, &quot;end&quot;))
# Add start
med_chr_lens$start = 1
# Reorder
med_chr_lens = med_chr_lens %&gt;% 
  dplyr::select(chr, start, end)
# Create custom genome
med_genome = regioneR::toGRanges(med_chr_lens)</code></pre>
</div>
<div id="load-exon-ranges" class="section level2">
<h2><span class="header-section-number">4.2</span> Load exon ranges</h2>
<pre class="r"><code># Load exon density for app
ex_ranges = readRDS(here(&quot;data&quot;, &quot;20201125_exon_ranges.rds&quot;))</code></pre>
</div>
<div id="make-palette" class="section level2">
<h2><span class="header-section-number">4.3</span> Make palette</h2>
<pre class="r"><code>electro_angler = c(&quot;#7400b8&quot;,&quot;#6930c3&quot;,&quot;#5e60ce&quot;,&quot;#5390d9&quot;,&quot;#4ea8de&quot;,&quot;#48bfe3&quot;,&quot;#56cfe1&quot;,&quot;#64dfdf&quot;,&quot;#72efdd&quot;,&quot;#80ffdb&quot;)
new_pal = grDevices::colorRampPalette(electro_angler)</code></pre>
</div>
</div>
<div id="generate-karyoplot" class="section level1">
<h1><span class="header-section-number">5</span> Generate karyoplot</h1>
<div id="full-genome" class="section level2">
<h2><span class="header-section-number">5.1</span> Full genome</h2>
<p><strong>NOTE</strong>: The y-scale for each class of repeat element is different.</p>
<pre class="r"><code># Create Karyoplot
kp = plotKaryotype(med_genome, chromosomes = NULL)
# Add base numbers 
karyoploteR::kpAddBaseNumbers(kp, tick.dist = 5000000, minor.tick.dist = 1000000, add.units = T, cex = 0.5)

# Add mirror exon density to ideogram
kpPlotDensity(kp, data=ex_ranges, col = &quot;#f77cb5&quot;,
              data.panel = &quot;ideogram&quot;,
              window.size = 25000,
              r0 = 0.5, r1 = 1)
kpPlotDensity(kp, data=ex_ranges, col = &quot;#f77cb5&quot;,
              data.panel = &quot;ideogram&quot;,
              window.size = 25000,
              r0 = 0.5, r1 = 0)

# Set target classes
target_classes = c(&quot;DNA&quot;, &quot;LINE&quot;, &quot;LTR&quot;, &quot;SINE&quot;, &quot;misc&quot;)

# Get colours
cols = new_pal(length(target_classes))

# Create function to add densities to plot
for (i in seq_len(length(target_classes))){
  # get target_class
  target_class = target_classes[i]
  # get autotrack to automatically adjust r0 and r1
  at = autotrack(i, length(target_classes), r0=0, r1=1)
  # add densities
  kp = kpPlotDensity(kp, data = rep_list_ranges[[target_class]],
                     col = cols[i],
                     r0=at$r0, r1=at$r1,
                     window.size = 25000)
  computed_ymax = ceiling(kp$latest.plot$computed.values$max.density)
  # add axis
  kpAxis(kp, ymin=0, ymax=computed_ymax, numticks = 2,
         labels = c(NA, computed_ymax),
         r0=at$r0, r1=at$r1,
         cex = 0.5)
  # add labels
  kpAddLabels(kp, labels = target_class,
              r0=at$r0, r1=at$r1,
              cex = 0.6, label.margin = 0.035)
}</code></pre>
<p><img src="20201126_repeats_files/figure-html/unnamed-chunk-12-1.png" width="1632" /></p>
</div>
<div id="per-chromosome" class="section level2">
<h2><span class="header-section-number">5.2</span> Per chromosome</h2>
<p><strong>NOTE</strong>: As the plot for every chromosome is generated separately, the y-scale for the same class of repeat element changes with each plot. The chromosomes now also look as if they’re the same length. It is therefore preferable to scan for genome-wide anomolies using the full-genome plot above.</p>
<pre class="r"><code>for (chr in as.character(seq(1, 24))){
  # Create Karyoplot
  kp = plotKaryotype(med_genome, chromosomes = chr)
  # Add base numbers 
  karyoploteR::kpAddBaseNumbers(kp, tick.dist = 5000000, minor.tick.dist = 1000000, add.units = T, cex = 0.5)
  
  # Add mirror exon density to ideogram
  kpPlotDensity(kp, data=ex_ranges, col = &quot;#f77cb5&quot;,
                data.panel = &quot;ideogram&quot;,
                window.size = 25000,
                r0 = 0.5, r1 = 1)
  kpPlotDensity(kp, data=ex_ranges, col = &quot;#f77cb5&quot;,
                data.panel = &quot;ideogram&quot;,
                window.size = 25000,
                r0 = 0.5, r1 = 0)
  
  # Set target classes
  target_classes = c(&quot;DNA&quot;, &quot;LINE&quot;, &quot;LTR&quot;, &quot;SINE&quot;, &quot;misc&quot;)
  
  # Get colours
  cols = new_pal(length(target_classes))
  
  # Create function to add densities to plot
  for (i in seq_len(length(target_classes))){
    # get target_class
    target_class = target_classes[i]
    # get autotrack to automatically adjust r0 and r1
    at = autotrack(i, length(target_classes), r0=0, r1=1)
    # add densities
    kp = kpPlotDensity(kp, data = rep_list_ranges[[target_class]],
                       col = cols[i],
                       r0=at$r0, r1=at$r1,
                       window.size = 25000)
    computed_ymax = ceiling(kp$latest.plot$computed.values$max.density)
    # add axis
    kpAxis(kp, ymin=0, ymax=computed_ymax, numticks = 2,
           labels = c(NA, computed_ymax),
           r0=at$r0, r1=at$r1,
           cex = 0.5)
    # add labels
    kpAddLabels(kp, labels = target_class,
                r0=at$r0, r1=at$r1,
                cex = 0.6, label.margin = 0.035)
  }
}  </code></pre>
<p><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-1.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-2.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-3.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-4.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-5.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-6.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-7.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-8.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-9.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-10.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-11.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-12.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-13.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-14.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-15.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-16.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-17.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-18.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-19.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-20.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-21.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-22.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-23.png" width="1440" /><img src="20201126_repeats_files/figure-html/unnamed-chunk-13-24.png" width="1440" /></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
